src/ft2font_wrapper.cpp

```diff
@@ -488,12 +490,14 @@ static int PyFT2Font_init(PyFT2Font *self, PyObject *args, PyObject *kwds)
                || !PyBytes_Check(data)) {
         PyErr_SetString(PyExc_TypeError,
                         "First argument must be a path or binary-mode file object");
+        ret = -1;
         goto exit;
     } else {
         self->py_file = filename;
         self->stream.close = NULL;
         Py_INCREF(filename);
     }
+    Py_CLEAR(data);
```

Comment:
Isn't the Py_XDECREF call in the exit: block enough?

Reply:
No, because the `CALL_CPP*` macros below this can exit early without going through the `exit` label.

Reply:
Then perhaps Py_XDECREF here too?  I have to admit the distinction between the two is a bit fuzzy for me (despite reading the cpython docs about them).
Also please add a comment explaining why it needs to be cleared/decrefed here, the nonlocal exit from the macro is just... nasty :/

Reply:
`Py_CLEAR` also sets `data` to `NULL`, so the `Py_XDECREF` below doesn't decrement it again. Alternatively, we could just move `Py_XDECREF` into the above `if`; it's only important in one case.

Reply:
ah yes, I see, I guess I prefer just moving things up then.